packages/code-editor/src/Markdown/utils.ts

```diff
@@ -0,0 +1,138 @@
+/*
+
+ MIT License
+
+ Copyright (c) 2021 Looker Data Sciences, Inc.
+
+ Permission is hereby granted, free of charge, to any person obtaining a copy
+ of this software and associated documentation files (the "Software"), to deal
+ in the Software without restriction, including without limitation the rights
+ to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ copies of the Software, and to permit persons to whom the Software is
+ furnished to do so, subject to the following conditions:
+
+ The above copyright notice and this permission notice shall be included in all
+ copies or substantial portions of the Software.
+
+ THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ SOFTWARE.
+
+ */
+
+/**
+ * TODO: Upgrade build to esnext and use `replaceAll`
+ */
+export const regReplaceAll = (
+  content: string,
+  pattern: string,
+  replacementFunc: any
+) => {
+  const target = new RegExp(pattern, 'gi')
+  return content.replace(target, replacementFunc)
+}
+/**
+ * Adds <mark> tags around text if search pattern is detected
+ * @param content - the main content to render
+ * @param searchPattern - the search pattern to consider
+ * @returns - a 'marked' string to be rendered by markdown component
+ */
+export const addMarkTags = (content: string, searchPattern: string) => {
+  let markedContent
+  try {
+    const replacement = (match: string) => `<mark>${match}</mark>`
+    markedContent = regReplaceAll(content, searchPattern, replacement)
+  } catch (e) {
+    markedContent = content
+  }
+  return markedContent
+}
+
+/**
+ * Removes <mark><mark/> tags from markdown text. The mark tag is removed from code styled
+ * text as the code renderer cannot differentiate <mark><mark/> from the code text.
+ * @param markedText - the markdown text input that contains <mark /> tags
+ */
+const removeMarkTags = (markedText: string) => {
+  return markedText.replace(/<mark>/g, '').replace(/<\/mark>/g, '')
```

Comment:
I think we can use this and it should run faster
```suggestion
  return markedText.replace(/<\/?mark>/g, '')
```